docs/api-tutorials/custom-reporter.md

```diff
@@ -0,0 +1,119 @@
+        this.decreaseIndent();
+      })
+      .on(EVENT_TEST_PASS, test => {
+        // Test#fullTitle() returns the suite name(s)
+        // prepended to the test title
+        console.log(`${this.indent()}pass: ${test.fullTitle()}`);
+      })
+      .on(EVENT_TEST_FAIL, (test, err) => {
+        console.log(
+          `${this.indent()}fail: ${test.fullTitle()} - error: ${err.message}`
+        );
+      })
+      .once(EVENT_RUN_END, () => {
+        console.log(
+          `end: ${this.stats.passes}/${this.stats.passes +
+            this.stats.failures} ok`
+        );
+      });
+  }
+
+  indent() {
+    return Array(this._indents)
+      .fill('  ')
+      .join('');
+  }
+
+  increaseIndent() {
+    this._indents++;
+  }
+
+  decreaseIndent() {
+    this._indents--;
+  }
+}
+
+module.exports = MyReporter;
+```
+
+To use this reporter, execute `mocha --reporter /path/to/my-reporter.js`.
+
+For further examples, the built-in reporter implementations are the [best place to look](https://github.com/mochajs/mocha/tree/master/lib/reporters). The [integration tests](https://github.com/mochajs/mocha/tree/master/test/reporters) may also be helpful.
+
+## The `Base` Reporter Class
+
+[Base] is an "abstract" class.  It contains console-specific settings and utilities, commonly used by built-in reporters.  Browsing the built-in reporter implementations, you may often see static properties of [Base] referenced.
+
+It's often useful--but not necessary--for a custom reporter to extend [Base].
+
+## Statistics
+
+Mocha adds a `stats` property of type [StatsCollector](/api/module-lib_stats-collector.html) to each reporter instance immediately following instantiaton.
+
+## Events
+
+A reporter should listen for events emitted from the `runner` (a singleton instance of [Runner]).
+
+The event names are exported from the `constants` property of `Mocha.Runner`:
+
+| Constant             | Event Name  | Event Arguments | Description                                                                                                                                                   |
+| -------------------- | ----------- | --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `EVENT_RUN_BEGIN`    | `start`     | _(n/a)_         | Execution will begin. Corresponding "end" event is `EVENT_RUN_END`.                                                                                           |
+| `EVENT_RUN_END`      | `end`       | _(n/a)_         | All [Suite]s, [Test]s and [Hook]s have completed execution. Corresponding "begin" event is `EVENT_RUN_BEGIN`.                                                 |
+| `EVENT_RUN_DELAY`    | `waiting`   | _(n/a)_         | Waiting for `global.run()` to be called; only emitted when `delay` option is `true`.                                                                          |
```

Comment:
While I understand what you're saying here, a link to [Delayed Root Suite](https://mochajs.org/#delayed-root-suite) might be appropriate...

Reply:
- [ ] link to delayed root suite docs